testing/testcafe/tests/editors/overlays/scrolling.ts

```diff
@@ -0,0 +1,145 @@
+          const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+          const overlay = new Popup('#overlay');
+
+          const body = Selector('body');
+
+          const checkBodyStyles = async ({ paddingRight, overflow }) => {
+            await t
+              .expect(getComputedPropertyValue('body', 'padding-right'))
+              .eql(paddingRight)
+              .expect(getComputedPropertyValue('body', 'overflow'))
+              .eql(overflow)
+              .expect(getComputedPropertyValue('body', 'position'))
+              .eql('static')
+              .expect(getComputedPropertyValue('body', 'top'))
+              .eql('auto')
+              .expect(getComputedPropertyValue('body', 'left'))
+              .eql('auto');
+          };
+
+          await t
+            .scroll(body, 300, 0);
+
+          await checkBodyStyles({ paddingRight: '0px', overflow: 'visible' });
+          await t
+            .expect(getDocumentScrollTop())
+            .eql(258);
+
+          await insertStylesheetRulesToPage('body { padding-right: 10px; overflow: auto;}');
+
+          await checkBodyStyles({ paddingRight: '10px', overflow: 'auto' });
+          await t
+            .expect(getDocumentScrollTop())
+            .eql(258);
+
+          await overlay.show();
+
+          await checkBodyStyles({ paddingRight: enableBodyScroll ? '10px' : '25px', overflow: enableBodyScroll ? 'auto' : 'hidden' });
+          await t
+            .expect(getDocumentScrollTop())
+            .eql(258);
+
+          await testScreenshot(t, takeScreenshot, `Overlay native scrolling,shading=${shading},enableBodyScroll=${enableBodyScroll},fullScreen=${fullScreen}.png`);
+
+          await overlay.hide();
+
+          await checkBodyStyles({ paddingRight: '10px', overflow: 'auto' });
+          await t
+            .expect(getDocumentScrollTop())
+            .eql(258);
+
+          await t
+            .expect(compareResults.isValid())
+            .ok(compareResults.errorMessages());
+        }).before(async () => {
+          await appendElementTo('#container', 'div', 'scrollable-container', { height: '2000px', overflowY: 'auto' });
+          await appendElementTo('#scrollable-container', 'div', 'scrollable-content', { height: '3000px' });
+
+          await appendElementTo('#scrollable-content', 'div', 'inner-container', {
+            width: '500px', height: '500px', border: '1px solid black', overflow: 'auto',
+          });
+
+          await ClientFunction(() => {
+            const $content = $('<div>');
+
+            for (let i = 0; i < 100; i += 1) {
+              $content.append(`<div>${i}</div>`);
+            }
+
+            $('#scrollable-content').append($content);
+          })();
+
+          await appendElementTo('#inner-container', 'div', 'inner-content', { width: '2000px', height: '2000px' });
+          await appendElementTo('#scrollable-container', 'div', 'overlay', {});
+
+          await createWidget('dxOverlay', {
```

Comment:
I don't think we should test a UI of private Overlay component.
Moreover, `fullScreen` property is useless for it - it's implemented on a Popup level.
Let's use a Popup for a screenshot test?

Reply:
Yes, great idea.